syntax = "proto3";

service ImmVis {
    rpc OpenDatasetFile(OpenDatasetFileRequest) returns (OpenDatasetFileResponse) {}
    rpc GetDatasetDimensions(Void) returns (stream DimensionInfo) {}
    rpc GetDimensionFloatValues(GetDimension) returns (stream FloatDimensionValue) {}
}

message Void {}

message OpenDatasetFileRequest {
    string filePath = 1;
}

message OpenDatasetFileResponse {
    int32 responseCode = 1;
}

message DimensionInfo {
    string name = 1;
    string type = 2;
}

message GetDimension {
    string dimensionName = 1;
}

message FloatDimensionValue {
    float value = 1;
}